src/shared/modules/auth/common.js

```diff
@@ -0,0 +1,217 @@
+        )
+        currentSSOProviders.splice(idx, 1)
+        authLog(`Updating SSO provider with id: "${provider.id}"`)
+      } else {
+        authLog(
+          `Not accepting discovered SSO provider with id: "${provider.id}", id exists already`
+        )
+        return
+      }
+    }
+    currentSSOProviders.push(provider)
+  })
+
+  window.sessionStorage.setItem(
+    AUTH_STORAGE_SSO_PROVIDERS,
+    JSON.stringify(currentSSOProviders)
+  )
+  authLog('Checked and merged SSO providers')
+}
+
+export const getSSOProvidersFromStorage = () => {
+  const ssoProviders = JSON.parse(
+    window.sessionStorage.getItem(AUTH_STORAGE_SSO_PROVIDERS)
+  )
+  if (!ssoProviders || !ssoProviders.length) {
+    authLog('No SSO providers in (local) storage found')
+    return []
+  }
+  if (!window.isSecureContext) {
+    authLog(
+      'This application is NOT executed in a secure context. SSO support is therefore disabled. Load the application in a secure context to proceed with SSO.',
+      'warn'
+    )
+    return []
+  }
+  return ssoProviders
+}
+
+export const getSSOProviderByIdpId = idpId => {
+  const ssoProviders = getSSOProvidersFromStorage()
+
+  const selectedSSOProvider = ssoProviders.find(
+    provider => provider.id === idpId
+  )
+  if (!selectedSSOProvider) {
+    authLog(`No matching SSO provider to passed in argument: ${idpId}`, 'warn')
+    return null
+  }
+  return selectedSSOProvider
+}
+
+export const getCredentialsFromAuthResult = (result, idpId) => {
+  const emptyCredentials = { username: '', password: '' }
+  authLog(`Attempting to assemble credentials for idp_id: ${idpId}`)
+
+  if (!result || !idpId) {
+    authLog('No result or idp_id passed in', 'warn')
+    return emptyCredentials
+  }
+
+  const selectedSSOProvider = getSSOProviderByIdpId(idpId)
+  if (!selectedSSOProvider) return emptyCredentials
+
+  const tokenTypePrincipal =
+    selectedSSOProvider.config?.['token_type_principal'] ||
+    defaultTokenTypePrincipal
+  authLog(
+    `Credentials, using token type "${tokenTypePrincipal}" to retrieve principal`
+  )
+
+  const parsedJWT = jwtDecode(result[tokenTypePrincipal])
+  authDebug('Credentials, parsed JWT', parsedJWT)
+
+  if (!parsedJWT) {
+    authLog(
+      `Could not parse JWT of type "${tokenTypePrincipal}" for idp_id "${idpId}", aborting`,
+      'warn'
+    )
+    return emptyCredentials
+  }
+
+  const principal = selectedSSOProvider.config?.principal || ''
+  authLog(`Credentials, provided principal in config: ${principal}`)
+
+  const credsPrincipal =
+    parsedJWT[principal] || parsedJWT.email || parsedJWT.sub
+  authLog(`Credentials assembly with username: ${credsPrincipal}`)
+
+  const tokenTypeAuthentication =
+    selectedSSOProvider.config?.['token_type_authentication'] ||
+    defaultTokenTypeAuthentication
+  authLog(
+    `Credentials assembled with token type "${tokenTypeAuthentication}" as password. If connection still does not succeed, make sure neo4j.conf is set up correctly`
```

Comment:
should this also log whether `result[tokenTypeAuthentication]` had a value or not?